shepherd_bms/include/datastructs.h

```diff
@@ -1,5 +1,66 @@
 #ifndef DATASTRUCTS_H
 #define DATASTRUCTS_H
 
+#include <nerduino.h>
+
+
+
+/*Individual chip data*/
+typedef struct ChipData_t
+{
+    uint16_t VoltageReading[12];          //store voltage readings from each chip
+    uint16_t thermistorReading[32];       //store all therm readings from each chip
+    bool discharge[12];
+};
+
+typedef enum BMSFault_t
+{
+    //Orion BMS faults
+    CELLS_NOT_BALANCING;
+    CELL_VOLTAGE_TOO_HIGH;
+    CELL_VOLTAGE_TOO_LOW;
+    CELL_ASIC_FAULT;                    //internal fault with the cell voltage measurement circuitry
+    PACK_TOO_HOT;
+    CHARGE_INTERLOCK_FAULT;             //both Charge Power (pin 3) and Ready Power (pin 2) are energized at the same time
```

Comment:
```suggestion

```

Reply:
We probably wont have two power pins for charging

Reply:
Technically we have a digital buffer reading the charging status, so we can have a fault like "internal state of charging does not match the reading of the charge line"
![image](https://user-images.githubusercontent.com/80368116/197368508-b6395a11-7f3f-41b0-9c4f-27a40fe772b2.png)